Boa.Constrictor/Dumping/Dumpers/TestLogReportDumper.cs

```diff
@@ -0,0 +1,288 @@
+﻿using Boa.Constrictor.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Boa.Constrictor.Dumping
+{
+    /// <summary>
+    /// Dumps an HTML report that combines all test logs.
+    /// </summary>
+    public class TestLogReportDumper : AbstractDumper
+    {
+        #region Constants
+
+        /// <summary>
+        /// The default title for the report.
+        /// </summary>
+        public const string DefaultTitle = "Test Log Report";
+
+        /// <summary>
+        /// The HTML file extension.
+        /// </summary>
+        public const string HtmlExtension = ".html";
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// The report title.
+        /// </summary>
+        public string Title { get; private set; }
+        
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="name">A descriptive name for the dumper.</param>
+        /// <param name="dumpDir">The output directory for dumping requests and responses.</param>
+        /// <param name="fileToken">The token for the file name.</param>
+        /// <param name="title">The report title.</param>
+        public TestLogReportDumper(string name, string dumpDir, string fileToken, string title = DefaultTitle) :
+            base(name, dumpDir, fileToken) => Title = title;
+
+        #endregion
+
+        #region Private Methods
+        
+        /// <summary>
+        /// Gets CSS text for a result's background color.
+        /// String format: "background-color: #xxxxxx;".
+        /// Returns an empty string if no color is designated for the result.
+        /// </summary>
+        /// <param name="result">The result string.</param>
+        /// <returns></returns>
+        private string GetBackgroundColor(string result)
+        {
+            string lowerResult = result.ToLower();
+            string background = "";
+
+            if (lowerResult.Contains("pass") || lowerResult.Contains("success") || lowerResult.Contains("ok"))
```

Comment:
NIT: Could add comment here to indicate what colors these are, or even create constants above named to indicate the colors